adal/oauth2_client.py

```diff
@@ -138,7 +138,7 @@ def _parse_id_token(self, encoded_token):
                 self._log.warn('The returned id_token could not be base64 url safe decoded.')
                 return
 
-            id_token = json.loads(b64_decoded.decode())
+            id_token = json.loads(b64_decoded.decode('utf-8'))

```

Comment:
Yea now I also noticed that [the Travis test failure](https://travis-ci.org/AzureAD/azure-activedirectory-library-for-python/jobs/171546037#L478) is caused by the `raise` improvement.

Adding a `with self.assertRaises(UnicodeDecodeError):` into that new problematic `test_bad_id_token_base64_in_response()` test case seems to do the trick in terms of getting the test cases pass.